### PR TITLE
Bug 1805981: oc rsh to an etcd pod needs to have etcdctl "just work" 

### PR DESCRIPTION
**Pull request by sanchezl**

### CODE REVIEW

File: bindata/etcd/pod.yaml [hunk 1]
```diff
@@ -37,6 +37,31 @@ spec:
         - mountPath: /etc/kubernetes/static-pod-certs
           name: cert-dir
   containers:
+  - name: etcdctl
```
Comment (deads2k) [inline bindata/etcd/pod.yaml]: comment explaining why this container must be first or we'll move it.

File: pkg/operator/targetconfigcontroller/etcd_env.go [hunk 2]
```diff
@@ -72,11 +73,24 @@ func getFixedEtcdEnvVars(envVarContext envVarContext) (map[string]string, error)
...
+	envVars, err := getAllClusterMembers(envVarContext)
+	if err != nil {
+		return nil, err
+	}
+	return map[string]string{
+		"ETCDCTL_API":       "3",
+		"ETCDCTL_CACERT":    "/etc/kubernetes/static-pod-resources/configmaps/etcd-serving-ca/ca-bundle.crt",
+		"ETCDCTL_CERT":      "/etc/kubernetes/static-pod-resources/secrets/etcd-all-peer/etcd-peer-NODE_NAME.crt",
+		"ETCDCTL_KEY":       "/etc/kubernetes/static-pod-resources/secrets/etcd-all-peer/etcd-peer-NODE_NAME.key",
+		"ETCDCTL_ENDPOINTS": envVars["ALL_ETCD_ENDPOINTS"],
```
Comment (deads2k) [inline pkg/operator/targetconfigcontroller/etcd_env.go]: This has no impact on the `etcd` process itself?

File: pkg/operator/targetconfigcontroller/etcd_env.go [hunk 3]
```diff
@@ -72,11 +73,24 @@ func getFixedEtcdEnvVars(envVarContext envVarContext) (map[string]string, error)
 	return map[string]string{
 		"ETCD_DATA_DIR":              "/var/lib/etcd",
 		"ETCD_QUOTA_BACKEND_BYTES":   "7516192768", // 7 gig
-		"ETCDCTL_API":                "3",
 		"ETCD_INITIAL_CLUSTER_STATE": "existing",
 	}, nil
 }
 
+func getEtcdctlEnvVars(envVarContext envVarContext) (map[string]string, error) {
+	envVars, err := getAllClusterMembers(envVarContext)
```
Comment (deads2k) [inline pkg/operator/targetconfigcontroller/etcd_env.go]: no.  Refactor into a method that gives just the string and use it in both.

File: pkg/operator/targetconfigcontroller/etcd_env.go (hunk 2 above)
Comment (hexfusion) [reply pkg/operator/targetconfigcontroller/etcd_env.go]: any var that is prefixed with `ETCDCTL_` is only consumed by the client binary. Where `ETCD_` is consumed by server instance.

**Pull request merged**